File: benchmark/lua/bubble.lua
local n = 500
local arr = {}
for i = n, 1, -1 do
    table.insert(arr, i)
end
local start = os.clock()
for k = 1, n do
    for j = 1, n - 1 do
        if arr[j] > arr[j + 1] then
            local temp = arr[j]
            arr[j] = arr[j + 1]
            arr[j + 1] = temp
        end
    end
end
local finish = os.clock()
print("First: " .. arr[1])
print("Last: " .. arr[n])
print(string.format("Time: %.0fms", (finish - start) * 1000))

File: benchmark/python/bubble.py
import time

n = 500
arr = []
for i in range(n, 0, -1):
    arr.append(i)
start = time.time() * 1000
for k in range(n):
    for j in range(n - 1):
        if arr[j] > arr[j + 1]:
            temp = arr[j]
            arr[j] = arr[j + 1]
            arr[j + 1] = temp
end = time.time() * 1000
print(f"First: {arr[0]}")
print(f"Last: {arr[n - 1]}")
print(f"Time: {end - start:.0f}ms")

